projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { TestBed, waitForAsync } from '@angular/core/testing';
+import { FhirDataService } from './fhir-data.service';
+import FHIR from 'fhirclient';
+import { fhirclient } from 'fhirclient/lib/types';
+import { cold, getTestScheduler } from 'jasmine-marbles';
+
+describe('FhirDataService', () => {
+  let service: FhirDataService;
+
+  beforeEach(() => {
+    sessionStorage.clear();
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(FhirDataService);
+  });
+
+  afterEach(() => {
+    sessionStorage.clear();
+  });
+
+  describe('initialize', () => {
+    it('should create a client from EHR launch context', async () => {
+      await FHIR.oauth2.authorize({
+        fhirServiceUrl: 'http://example.com/ehr',
+        completeInTarget: true,
+        noRedirect: true,
+        fakeTokenResponse: {
+          serverUrl: 'http://example.com/ehr',
+        },
+      });
+      await service.initialize();
+      expect(service.client?.getState('serverUrl')).toEqual('http://example.com/ehr');
+    });
+
+    it('should create a client from clientState parameter when there is no EHR launch context', async () => {
+      await service.initialize({
+        serverUrl: 'http://example.com/open',
+      });
+      expect(service.client?.getState('serverUrl')).toEqual('http://example.com/open');
+    });
+  });
+
+  describe('getPatientData', () => {
+    beforeEach(async () => {
+      await service.initialize({
+        serverUrl: 'http://example.com/open',
+        tokenResponse: {
+          patient: '123',
+        },
+      });
+      if (!service.client) {
+        throw new Error('Client should be initialized');
+      }
+    });
+
+    it('should query the FHIR server on subscribe', waitForAsync(() => {
+      const request = spyOn(service.client!.patient, 'request').and.resolveTo(null);
+      const bundles$ = service.getPatientData('Observation');
+      bundles$.subscribe({
+        complete: () => {
+          expect(request).toHaveBeenCalledWith(jasmine.stringMatching(/^Observation/), jasmine.anything());
```

Comment:
This test case is just making sure that `getPatientData` makes a call to `Client.patient.request()` for the correct resource type (which, in real code, will query the FHIR server). That expectation is checking the parameters that were passed in to the `Client.patient.request()` spy from `getPatientData`:
1. The first parameter should start with "Observation". I used a regex instead of exact string match because `getPatientData` may add a query string (`?_count=100`) and I didn't want this test to care about page size.
2. The second parameter can be anything. I didn't think it was worth checking the `fhirOptions` config object here. `onPage` callback is covered by other tests